envoy/datadog_checks/envoy/envoy.py

```diff
@@ -0,0 +1,78 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import requests
+
+from datadog_checks.checks import AgentCheck
+
+from .errors import UnknownMetric
+from .parser import parse_metric
+
+
+class Envoy(AgentCheck):
+    SERVICE_CHECK_NAME = 'envoy.can_connect'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        super(Envoy, self).__init__(name, init_config, agentConfig, instances)
+        self.unknown_metrics = set()
+
+    def check(self, instance):
+        custom_tags = instance.get('tags', [])
+
+        try:
+            stats_url = instance['stats_url']
+        except KeyError:
+            msg = 'Envoy configuration setting `stats_url` is required'
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, message=msg, tags=custom_tags)
+            self.log.error(msg)
+            return
+
+        username = instance.get('username', None)
+        password = instance.get('password', None)
+        auth = (username, password) if username and password else None
+        verify_ssl = instance.get('verify_ssl', True)
+        proxies = self.get_instance_proxy(instance, stats_url)
+        timeout = int(instance.get('timeout', 20))
+
+        try:
+            request = requests.get(
+                stats_url, auth=auth, verify=verify_ssl, proxies=proxies, timeout=timeout
+            )
+        except requests.exceptions.Timeout:
+            msg = 'Envoy endpoint `{}` timed out after {} seconds'.format(stats_url, timeout)
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, message=msg, tags=custom_tags)
+            self.log.exception(msg)
+            return
+        except (requests.exceptions.RequestException, requests.exceptions.ConnectionError):
+            msg = 'Error accessing Envoy endpoint `{}`'.format(stats_url)
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, message=msg, tags=custom_tags)
+            self.log.exception(msg)
+            return
+
+        if request.status_code != 200:
+            msg = 'Envoy endpoint `{}` responded with HTTP status code {}'.format(stats_url, request.status_code)
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, message=msg, tags=custom_tags)
+            self.log.warning(msg)
+            return
+
+        get_method = getattr
```

Comment:
just curious: why this alias?

Reply:
To avoid a global lookup many times during the loop.

Reply:
Ok for me, mind to add a comment above that line? So it doesn't get wiped by mistake in the future.